src/reducers/productsReducer.js

```diff
@@ -0,0 +1,69 @@
+import { useReducer } from "react";
+import { productsData } from "../data/product-data";
+
+export default function useProductsReducer() {
+  const initialState = {
+    productData: productsData,
+    price: 50,
+    rating: 1,
+    sort: null,
+    category: [],
+    bird: [],
+  };
+
+  const [state, dispatch] = useReducer((state, action) => {
+    switch (action.type) {
+      case "SETVALUE":
+        return {
+          ...state,
+          [action.actionKey]: action.actionValue,
+        };
+      case "FILTER":
+        return {
+          ...state,
+          [action.filterType]: state[action.filterType].includes(action.filter)
+            ? state[action.filterType].filter((type) => type !== action.filter)
+            : [...state[action.filterType], action.filter],
+        };
+      case "RESET_FILTERS":
+        return initialState;
+      default:
+        return state;
```

Comment:
Suggestion : you can do error handling in the default case

Reply:
Done! Thanks for pointing it out.